ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/SandboxIndexAndWriteService.scala

```diff
@@ -345,11 +348,20 @@ private class SandboxIndexAndWriteService(
   }
 
   // IndexPackagesService
-  override def listPackages(): Future[Set[PackageId]] =
-    templateStore.listPackages()
+  override def listLfPackages(): Future[Map[PackageId, PackageDetails]] =
+    packageStore.listLfPackages()
 
-  override def getPackage(packageId: PackageId): Future[Option[Archive]] =
-    templateStore.getPackage(packageId)
+  override def getLfArchive(packageId: PackageId): Future[Option[Archive]] =
+    packageStore.getLfArchive(packageId)
+
+  override def getLfPackage(packageId: PackageId): Future[Option[Ast.Package]] =
+    packageStore.getLfPackage(packageId)
+
+  // PackageWriteService
+  override def uploadDar(
+      sourceDescription: String,
+      payload: Array[Byte]): CompletionStage[UploadDarResult] =
+    packageStore.uploadDar(ledger.getCurrentTime(), sourceDescription, payload)
```

Comment:
Instead of adding the `getCurrentTime` method to `Ledger` which acts as a simple delegation to `TimeProvider`, we can also pass the `timeProvider`, which is already passed to the factory methods, also to `SandboxIndexAndWriteService`. What do you think?